runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java

```diff
@@ -138,8 +145,10 @@
 	private static final int MAX_AMULET_OF_CHEMISTRY_CHARGES = 5;
 	private static final int MAX_AMULET_OF_BOUNTY_CHARGES = 10;
 	private static final int MAX_SLAYER_BRACELET_CHARGES = 30;
+	private static final int MAX_BLOOD_ESSENCE_CHARGES = 1000;
 
 	private int lastExplorerRingCharge = -1;
+	private boolean didBloodEssenceBreak = false;
```

Comment:
You can probably check for the text "You activate the blood essence." and set the charges to 1000, instead of using this boolean variable.